File: ncPersistencia/pEfdUnidadeMedida.vb
Imports ncDados.nsEFD
Imports ncComum.nsAcessoBD
Imports ncComum.nsFuncoes.cFuncoes
Imports ncComum.nsExcecao

Namespace nsEFD

  Public Class pEfdUnidadeMedida

    Public Function Listar() As ColecaoEfdUnidadeMedida

      Dim retorno As ColecaoEfdUnidadeMedida
      Dim acessoBanco As cAcessoBD
      Dim ds As DataSet
      Dim dt As DataTable
      Dim row As DataRow
      Dim item As dEfdUnidadeMedida
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " Select codigo, descricao From EfdUnidadeMedida Order By descricao "

        ds = acessoBanco.ExecutarDS(comandoSQL)

        If Not ds Is Nothing Then
          If ds.Tables.Count > 0 Then
            dt = ds.Tables(0)

            If dt.Rows.Count > 0 Then
              retorno = New ColecaoEfdUnidadeMedida

              For Each row In dt.Rows
                item = New dEfdUnidadeMedida

                item.codigo = RetornarTexto(row("codigo"))
                item.descricao = RetornarTexto(row("descricao"))

                retorno.Add(item)
              Next
            Else
              retorno = Nothing
            End If
          Else
            retorno = Nothing
          End If
        Else
          retorno = Nothing
        End If

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Listar EfdUnidadeMedida [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Listar = retorno

    End Function

    Public Function Consultar(ByVal dados As dEfdUnidadeMedida) As dEfdUnidadeMedida

      Dim retorno As dEfdUnidadeMedida
      Dim acessoBanco As cAcessoBD
      Dim ds As DataSet
      Dim dt As DataTable
      Dim sqlSelect As String
      Dim sqlWhere As String
      Dim sqlFrom As String

      Try

        acessoBanco = New cAcessoBD

        sqlSelect = " Select codigo, descricao "
        sqlFrom = " From EfdUnidadeMedida "
        sqlWhere = String.Empty

        '-- codigo
        sqlWhere = MontarParametrosSQL(sqlWhere, dados.codigo, "codigo")

        '-- descricao
        sqlWhere = MontarParametrosSQL(sqlWhere, dados.descricao, "descricao")

        If Not sqlWhere.Equals(String.Empty) Then
          sqlWhere = " WHERE " & sqlWhere
        End If

        ds = acessoBanco.ExecutarDS(sqlSelect & " " & sqlFrom & " " & sqlWhere & " Order By descricao ")

        If Not ds Is Nothing Then
          If ds.Tables.Count > 0 Then
            dt = ds.Tables(0)

            If dt.Rows.Count > 0 Then
              retorno = New dEfdUnidadeMedida

              retorno.codigo = RetornarTexto(dt.Rows(0).Item("codigo"))
              retorno.descricao = RetornarTexto(dt.Rows(0).Item("descricao"))
            Else
              retorno = Nothing
            End If
          Else
            retorno = Nothing
          End If
        Else
          retorno = Nothing
        End If

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Consultar EfdUnidadeMedida [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Consultar = retorno

    End Function

    Public Function Excluir(ByVal dados As dEfdUnidadeMedida) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " DELETE FROM EfdUnidadeMedida " & _
            " WHERE codigo = " & PersistirTexto(dados.codigo)

        retorno = acessoBanco.ExecutarINT(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Excluir EfdUnidadeMedida [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Excluir = retorno

    End Function

    Public Function Incluir(ByVal dados As dEfdUnidadeMedida) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " INSERT INTO " & _
            " EfdUnidadeMedida ( codigo, descricao ) " & _
            " VALUES (" & _
            PersistirTexto(dados.codigo) & "," & _
            PersistirTexto(dados.descricao) & ")"

        retorno = acessoBanco.ExecutarCID(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Incluir EfdUnidadeMedida [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Incluir = retorno

    End Function

    Public Function Alterar(ByVal dados As dEfdUnidadeMedida) As Integer

      Dim retorno As Integer
      Dim acessoBanco As cAcessoBD
      Dim comandoSQL As String

      Try

        acessoBanco = New cAcessoBD

        comandoSQL = " UPDATE EfdUnidadeMedida SET " & _
            " descricao = " & PersistirTexto(dados.descricao) & _
            " WHERE " & _
            " codigo = " & dados.codigo

        retorno = acessoBanco.ExecutarINT(comandoSQL)

      Catch ex As Exception

        retorno = Nothing
        Throw New ExcecaoNascomercio("Erro em Alterar EfdUnidadeMedida [" & Me.ToString() & "] - " & ex.Message)

      End Try

      Alterar = retorno

    End Function

  End Class

End Namespace

File: ncRepositorios/pEfdUnidadeMedida.cs
using System;
using System.Data;
using ncNComum.nsAcessoBD;
using ncNComum.nsExcecao;
using static ncNComum.nsFuncoes.cFuncoes;
using nsEFD;

namespace ncPersistencia.nsEFD
{
    public class pEfdUnidadeMedida
    {
        public ColecaoEfdUnidadeMedida Listar()
        {
            ColecaoEfdUnidadeMedida retorno = null;
            try
            {
                var acessoBanco = new cAcessoBD();
                string comandoSQL = " Select codigo, descricao From EfdUnidadeMedida Order By descricao ";
                var ds = acessoBanco.ExecutarDS(comandoSQL);
                if (ds != null && ds.Tables.Count > 0)
                {
                    DataTable dt = ds.Tables[0];
                    if (dt.Rows.Count > 0)
                    {
                        retorno = new ColecaoEfdUnidadeMedida();
                        foreach (DataRow row in dt.Rows)
                        {
                            var item = new dEfdUnidadeMedida();
                            item.codigo = RetornarTexto(row["codigo"]);
                            item.descricao = RetornarTexto(row["descricao"]);
                            retorno.Add(item);
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                retorno = null;
                throw new ExcecaoNascomercio("Erro em Listar EfdUnidadeMedida [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }

        public dEfdUnidadeMedida Consultar(dEfdUnidadeMedida dados)
        {
            dEfdUnidadeMedida retorno = null;
            try
            {
                var acessoBanco = new cAcessoBD();
                string sqlSelect = " Select codigo, descricao ";
                string sqlFrom = " From EfdUnidadeMedida ";
                string sqlWhere = string.Empty;
                sqlWhere = MontarParametrosSQL(sqlWhere, dados.codigo, "codigo");
                sqlWhere = MontarParametrosSQL(sqlWhere, dados.descricao, "descricao");
                if (!sqlWhere.Equals(string.Empty))
                    sqlWhere = " WHERE " + sqlWhere;
                var ds = acessoBanco.ExecutarDS(sqlSelect + " " + sqlFrom + " " + sqlWhere + " Order By descricao ");
                if (ds != null && ds.Tables.Count > 0)
                {
                    DataTable dt = ds.Tables[0];
                    if (dt.Rows.Count > 0)
                    {
                        retorno = new dEfdUnidadeMedida();
                        retorno.codigo = RetornarTexto(dt.Rows[0]["codigo"]);
                        retorno.descricao = RetornarTexto(dt.Rows[0]["descricao"]);
                    }
                }
            }
            catch (Exception ex)
            {
                retorno = null;
                throw new ExcecaoNascomercio("Erro em Consultar EfdUnidadeMedida [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }

        public int Excluir(dEfdUnidadeMedida dados)
        {
            int retorno = 0;
            try
            {
                var acessoBanco = new cAcessoBD();
                string comandoSQL = " DELETE FROM EfdUnidadeMedida " +
                    " WHERE codigo = " + PersistirTexto(dados.codigo);
                retorno = acessoBanco.ExecutarINT(comandoSQL);
            }
            catch (Exception ex)
            {
                retorno = 0;
                throw new ExcecaoNascomercio("Erro em Excluir EfdUnidadeMedida [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }

        public int Incluir(dEfdUnidadeMedida dados)
        {
            int retorno = 0;
            try
            {
                var acessoBanco = new cAcessoBD();
                string comandoSQL = " INSERT INTO " +
                    " EfdUnidadeMedida ( codigo, descricao ) " +
                    " VALUES (" +
                    PersistirTexto(dados.codigo) + "," +
                    PersistirTexto(dados.descricao) + ")";
                retorno = acessoBanco.ExecutarCID(comandoSQL);
            }
            catch (Exception ex)
            {
                retorno = 0;
                throw new ExcecaoNascomercio("Erro em Incluir EfdUnidadeMedida [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }

        public int Alterar(dEfdUnidadeMedida dados)
        {
            int retorno = 0;
            try
            {
                var acessoBanco = new cAcessoBD();
                string comandoSQL = " UPDATE EfdUnidadeMedida SET " +
                    " descricao = " + PersistirTexto(dados.descricao) +
                    " WHERE " +
                    " codigo = " + dados.codigo;
                retorno = acessoBanco.ExecutarINT(comandoSQL);
            }
            catch (Exception ex)
            {
                retorno = 0;
                throw new ExcecaoNascomercio("Erro em Alterar EfdUnidadeMedida [" + ToString() + "] - " + ex.Message);
            }
            return retorno;
        }
    }
}
